### ISSUE TITLE
Implement spatial correlation metrics (Category B)

### ISSUE THREAD

**Issue by Trecek** [TalonT-Org/spectral-init#195]
## Summary

Implement the 5 spatial correlation metrics that make the MERFISH benchmark
unique. These metrics use physical tissue coordinates as ground truth to
evaluate how well UMAP embeddings preserve spatial structure — something no
standard DR benchmark can measure.

## Context

- Plan reference: `docs/merfish-umap-benchmark-plan.md` Batch 4, Issue 4.1
- Metric definitions: `docs/metrics/spatial-correlation.md`
- The pipeline (PR #192) and preprocessing (#193) are complete
- Category A metrics (trustworthiness, silhouette, Procrustes, pairwise corr)
  are already implemented in `generate_umap_comparisons.py`
- Category B metrics are NEW — they require spatial coordinates, which flow
  through the pipeline as `merfish_10k_spatial.npz`

### Module: `tests/visual_eval/spatial_metrics.py`

Pure numpy/scipy/sklearn module — no scanpy or anndata dependency.

#### Metric 1: Spatial Neighbor Agreement (SNA)

```python
def spatial_neighbor_agreement(
    spatial_coords: np.ndarray,  # (n, 2) physical x,y
    embedding: np.ndarray,       # (n, 2) UMAP coords
    k: int = 15,
) -> float:
```
For each cell, compute Jaccard similarity between its k-nearest spatial
neighbors and k-nearest embedding neighbors. Return the mean over all cells.
Range [0, 1], higher = better spatial preservation.

#### Metric 2: Spatial Distance Rank Correlation

```python
def spatial_distance_correlation(
    spatial_coords: np.ndarray,
    embedding: np.ndarray,
    sample_size: int = 5000,
    seed: int = 42,
) -> float:
```
Spearman rank correlation between pairwise distances in physical space vs
embedding space. Subsample to `sample_size` cells for tractability.
Range [-1, 1], higher = better.

#### Metric 3: Moran's I of Embedding Coordinates

```python
def morans_i(
    spatial_coords: np.ndarray,
    values: np.ndarray,          # (n,) one UMAP dimension
    k: int = 6,
) -> float:
```
Global Moran's I using k-nearest spatial neighbors as the weight matrix.
Call once per UMAP dimension, report max. Range [-1, 1], near 1 = spatially
smooth embedding.

#### Metric 4: CHAOS Score

```python
def chaos_score(
    spatial_coords: np.ndarray,
    labels: np.ndarray,          # (n,) cluster IDs
) -> float:
```
Mean 1-nearest-neighbor distance within each cluster in physical space,
weighted by cluster size. Lower = more spatially compact clusters.

#### Metric 5: PAS Score

```python
def pas_score(
    spatial_coords: np.ndarray,
    labels: np.ndarray,
    k: int = 10,
    threshold: float = 0.6,
) -> float:
```
Fraction of cells whose k spatial neighbors mostly disagree on cluster label.
Lower = better spatial coherence.

#### Convenience Function

```python
def compute_spatial_metrics(
    spatial_coords: np.ndarray,
    embedding: np.ndarray,
    labels: np.ndarray,
    k_sna: int = 15,
    k_morans: int = 6,
    k_pas: int = 10,
) -> dict:
```
Runs all 5 metrics and returns a dict with keys: `sna`, `spatial_dist_corr`,
`morans_i_max`, `morans_i_dim0`, `morans_i_dim1`, `chaos`, `pas`.

## Dependencies

- `numpy`, `scipy.spatial.distance`, `scipy.stats.spearmanr`
- `sklearn.neighbors.NearestNeighbors`
- No scanpy, anndata, or polars dependency

## Testing

- Include basic smoke tests in the module (runnable via
  `python -m tests.visual_eval.spatial_metrics` or as pytest functions)
- Test with small synthetic data (e.g., blobs with known spatial layout)
  to verify metric directionality (spatially coherent > random)
### API — Function Signatures

- **REQ-API-001:** Each metric function must accept numpy arrays as input (no AnnData, no scanpy objects).
- **REQ-API-002:** Each metric function must return a single float value.
- **REQ-API-003:** A `compute_spatial_metrics()` convenience function must exist that runs all 5 metrics and returns a dict.
- **REQ-API-004:** The module must have zero dependency on scanpy, anndata, or polars — only numpy, scipy, and sklearn.

### SNA — Spatial Neighbor Agreement

- **REQ-SNA-001:** SNA must compute the mean Jaccard similarity between each cell's k-nearest spatial neighbors and k-nearest embedding neighbors.
- **REQ-SNA-002:** SNA must use `sklearn.neighbors.NearestNeighbors` with `algorithm='kd_tree'` for the 2D spatial coordinates.
- **REQ-SNA-003:** SNA must default to `k=15` matching the UMAP `n_neighbors` parameter.

### DIST — Spatial Distance Correlation

- **REQ-DIST-001:** The metric must compute Spearman rank correlation between pairwise spatial distances and pairwise embedding distances.
- **REQ-DIST-002:** The metric must subsample to at most `sample_size` cells (default 5000) with `seed=42` for reproducibility.

### MI — Moran's I

- **REQ-MI-001:** Moran's I must be computed using k-nearest spatial neighbors (default k=6) as the binary weight matrix.
- **REQ-MI-002:** The convenience function must compute Moran's I for each UMAP dimension independently and report both per-dimension values and the max.

### CHAOS — CHAOS Score

- **REQ-CHAOS-001:** CHAOS must compute mean 1-nearest-neighbor distance within each cluster using physical coordinates.
- **REQ-CHAOS-002:** The score must be weighted by cluster size (cells per cluster).
- **REQ-CHAOS-003:** Clusters with fewer than 2 cells must be skipped.

### PAS — PAS Score

- **REQ-PAS-001:** PAS must compute the fraction of cells whose k spatial neighbors have a majority different cluster label.
- **REQ-PAS-002:** PAS must default to `k=10` and `threshold=0.6`.

### TEST — Smoke Tests

- **REQ-TEST-001:** The module must include at least one smoke test using synthetic data (e.g., sklearn `make_blobs` with spatial layout matching cluster identity).
- **REQ-TEST-002:** The smoke test must verify metric directionality: a spatially coherent embedding must score better than a random permutation on SNA, Moran's I, CHAOS, and PAS.
